From aa925121adc6e76e362d0a6f04e7db63e4032ef5 Mon Sep 17 00:00:00 2001
From: Sveinung Kvilhaugsvik <sveinung84@users.sourceforge.net>
Date: Tue, 23 May 2017 16:03:45 +0200
Subject: [PATCH 1/2] webperimental: introduce README.webperimental.

Move history and differences from classic to README.webperimental and
include it as the description file. It is getting to big for game.ruleset.

See hrm Feature #662934
---
 data/webperimental/README.webperimental | 69 +++++++++++++++++++++++++++++++
 data/webperimental/game.ruleset         | 73 ++-------------------------------
 2 files changed, 72 insertions(+), 70 deletions(-)
 create mode 100644 data/webperimental/README.webperimental

diff --git a/data/webperimental/README.webperimental b/data/webperimental/README.webperimental
new file mode 100644
index 0000000..13c9117
--- /dev/null
+++ b/data/webperimental/README.webperimental
@@ -0,0 +1,69 @@
+Webperimental was based on the classic ruleset (via the deprecated fcweb
+ruleset). It is more like classic than non web Freeciv's experimental
+ruleset.
+
+Differences compared to the classic ruleset are:
+- Diplomat available from Alphabet
+- AI players are created with random trait values
+- Known techs tree cannot have holes in it
+- Diplomats can steal gold from enemy capitals (needs Banking)
+- Spies can steal gold from enemy cities (needs Banking)
+- A higher percentage of a nation's gold can be stolen from its capital
+  (unless it's in Anarchy).
+- An even higher percentage of a nation's gold can be stolen from its
+  capital if it is communist.
+- The killstack server setting is disabled by default.
+  - unit stack blocks bribe/sabotage unit needs killstack.
+- Spies can steal parts of other player's world map.
+- Spies belonging to the owner of the Manhattan Project can nuke cities.
+  - agents and spymasters may even escape before the nuke detonates.
+- Explorers can establish embassies once Writing is discovered.
+  This action consumes the explorer.
+- Cities can be founded within enemy borders during war and
+  before contact.
+- The Zone of Control rules have been turned off.
+- There's only one barbarian nation, attacking from both land and sea.
+- A player that knows Electricity has no fog of war inside his borders.
+- Foot soldiers can peacefully expel foreign civilian units from your
+  terrain.
+- It is possible to establish an embassy while inside a ship.
+- The startunits are by default spread over the map.
+- The Leader unit has been added to the default startunits.
+- The Leader unit can destroy a domestic city. (Be careful when you scorch
+  the earth. If your Leader dies you lose he game.)
+- Killing an enemy Leader gives you a few of his cities, some of his gold
+  and some of his knowledge. Not all his cities will follow you. Some will
+  rather turn barbarian.
+- The Leader can heal light weight units belonging to you or to your allies.
+- Labels the year using AD/BC.
+- All units can be airlifted:
+  - Light weight units can be airlifted once Flight is known.
+  - Medium weight units can be airlifted once Advanced Flight is known.
+  - Heavy weight units can be airlifted once Fusion Power is known.
+  - To airlift heavy and medium weight units an airport is required both
+  in the source and in the destination city.
+- Cities that know Flight can airlift one (light weight) unit per turn. An
+  Airport allows a city to airlift one more unit per turn.
+- Default minimum distance between cities is 3.
+- Default map generator is `Fair islands`. It generates identical start
+  position islands for all players.
+- The Howitzer and the Battleship can bombard.
+  - bombardment may hurt all units at the target tile.
+  - bombardment can't kill a unit.
+  - bombardment can kill population in a city.
+  - webperimental's bombardment can target unit stacks 3 tiles away.
+  - webperimental's bombardment doesn't replace regular attack.
+- The Battleship is weak against Bomber planes.
+- Moving from a ship to the beach doesn't spend all movement points.
+- Airlifting to and from allied cities is allowed by default.
+- Land units sees further on top of Mountain tiles.
+- Railroads can't be used by enemy units by default. Regular roads can be
+  used by the enemy even if the restrictinfra server setting is enabled.
+  (The restrictinfra server setting is enabled by default.)
+- An alliance has more in game consequences:
+  - Caravan and Freight can help allies to complete wonders.
+  - Sun Tzu's War Academy gives its veteran bonus (combat and production)
+    to both its owner and to all his allies.
+  - J.S. Bach's Cathedral makes two unhappy citizens content in allied
+    cities too.
+  - A. Smith's Trading Co. pays for your allies' 1 upkeep buildings too.
diff --git a/data/webperimental/game.ruleset b/data/webperimental/game.ruleset
index bc8387e..064b572 100644
--- a/data/webperimental/game.ruleset
+++ b/data/webperimental/game.ruleset
@@ -19,7 +19,7 @@ format_version=20
 [ruledit]
 
 ; Which file to read description in from.
-;description_file = ""
+description_file = "webperimental/README.webperimental"
 
 [about]
 ; Ruleset name
@@ -31,74 +31,7 @@ name = _("Webperimental")
 ; Description of the ruleset
 summary = _("\
 This ruleset has experimental rules that works with the web-client.\
- It was based on the classic ruleset (via the deprecated fcweb ruleset).\
- It is more like classic than non web Freeciv's experimental ruleset.\n\
-\n\
-Differences compared to the classic ruleset are:\n\
-- Diplomat available from Alphabet\n\
-- AI players are created with random trait values\n\
-- Known techs tree cannot have holes in it\n\
-- Diplomats can steal gold from enemy capitals (needs Banking)\n\
-- Spies can steal gold from enemy cities (needs Banking)\n\
-- A higher percentage of a nation's gold can be stolen from its capital \
-(unless it's in Anarchy).\n\
-- An even higher percentage of a nation's gold can be stolen from its \
-capital if it is communist.\n\
-- The killstack server setting is disabled by default.\n\
-  - unit stack blocks bribe/sabotage unit needs killstack.\n\
-- Spies can steal parts of other player's world map. \n\
-- Spies belonging to the owner of the Manhattan Project can nuke cities.\n\
-  - agents and spymasters may even escape before the nuke detonates.\n\
-- Explorers can establish embassies once Writing is discovered. \
-This action consumes the explorer.\n\
-- Cities can be founded within enemy borders during war and \
-before contact.\n\
-- The Zone of Control rules have been turned off.\n\
-- There's only one barbarian nation, attacking from both land and sea.\n\
-- A player that knows Electricity has no fog of war inside his borders.\n\
-- Foot soldiers can peacefully expel foreign civilian units from your terrain.\n\
-- It is possible to establish an embassy while inside a ship.\n\
-- The startunits are by default spread over the map.\n\
-- The Leader unit has been added to the default startunits.\n\
-- The Leader unit can destroy a domestic city. (Be careful when you scorch \
-the earth. If your Leader dies you lose he game.)\n\
-- Killing an enemy Leader gives you a few of his cities, some of his gold \
-and some of his knowledge. Not all his cities will follow you. Some will \
-rather turn barbarian.\n\
-- The Leader can heal light weight units belonging to you or to your \
-allies.\n\
-- Labels the year using AD/BC.\n\
-- All units can be airlifted:\n\
-  - Light weight units can be airlifted once Flight is known.\n\
-  - Medium weight units can be airlifted once Advanced Flight is known.\n\
-  - Heavy weight units can be airlifted once Fusion Power is known.\n\
-  - To airlift heavy and medium weight units an airport is required both \
-in the source and in the destination city.\n\
-- Cities that know Flight can airlift one (light weight) unit per turn. An \
-Airport allows a city to airlift one more unit per turn.\n\
-- Default minimum distance between cities is 3.\n\
-- Default map generator is `Fair islands`. It generates identical start \
-position islands for all players.\n\
-- The Howitzer and the Battleship can bombard.\n\
-  - bombardment may hurt all units at the target tile.\n\
-  - bombardment can't kill a unit.\n\
-  - bombardment can kill population in a city.\n\
-  - webperimental's bombardment can target unit stacks 3 tiles away.\n\
-  - webperimental's bombardment doesn't replace regular attack.\n\
-- The Battleship is weak against Bomber planes.\n\
-- Moving from a ship to the beach doesn't spend all movement points.\n\
-- Airlifting to and from allied cities is allowed by default.\n\
-- Land units sees further on top of Mountain tiles.\n\
-- Railroads can't be used by enemy units by default. Regular roads can be \
-used by the enemy even if the restrictinfra server setting is enabled. \
-(The restrictinfra server setting is enabled by default.)\n\
-- An alliance has more in game consequences:\n\
-  - Caravan and Freight can help allies to complete wonders.\n\
-  - Sun Tzu's War Academy gives its veteran bonus (combat and production) \
-to both its owner and to all his allies.\n\
-  - J.S. Bach's Cathedral makes two unhappy citizens content in allied \
-cities too.\n\
-  - A. Smith's Trading Co. pays for your allies' 1 upkeep buildings too.\n\
+ The differences from classic are listed in full in README.webperimental.\
 \n\
 This ruleset is still in development. If playing it makes you think that \
 a rule should change go to the Freeciv-web board at \
@@ -108,7 +41,7 @@ webperimental ruleset feed back and suggestion thread.\
 
 ; Detailed description
 ; When updating this, update also desciption_file in [ruledit] section to match
-; description = ""
+description = *webperimental/README.webperimental*
 
 ; What capabilities ruleset provides for the scenarios.
 ; mimimum-default-sets - Default units, terrains, buildings, etc
-- 
2.1.4

